iFixit/iFixitAndroid
# Google Login: Migrate to OAuth

Google login through OpenID has been deprecated for a little while now so
we have to migrate to Google OAuth. This results in a much better user
experience because it goes through the native interface and doesn't require
the user to enter their credentials.

This also increases the `minSdkVersion` to `9` which is android 2.3. This is
required for Google Play Services which provides the Google Login
functionality. Android 2.2 only makes up 0.65% of installs currently. It's
unfortunate to lose them but the old app will still work for them.

QA
==

 - Check Google login on a few places that require it:
   - Media manager
   - Favorite guides
   - Favoriting a guide
   - etc.
 - Test on various devices:
   - Android 2.3
   - Android 5+
   - Tablet
   - Phone
 - Test in both orientations:
   - Portrait
   - Landscape
 - Unexpected actions:
   - Decline permissions
   - Cancel dialogs
   - Change orientations
 - Test on various accounts:
   - No associated iFixit account
   - iFixit account with permissions accepted through the web signin
   - iFixit account after prior sign in on Android

Closes #249
